gitbook/audio-streams.md

```diff
@@ -2,143 +2,129 @@
 
 ## Audio stream? What's that?
 
-An audio stream is similar to music (remember the [Music]({{book.api}}/Music.html) class?). It has almost the same functions and behaves the same. The only difference is that an audio stream doesn't play an audio file: Instead, it plays a custom audio source that *you* directly provide. In other words, defining your own audio stream allows you to play from more than just a file: A sound streamed over the network, music generated by your program, an audio format that SFML doesn't support, etc. 
+An audio stream is similar to music (remember the [Music]({{book.api}}/Music.html) class?). It has almost the same functions and behaves the same. The only difference is that an audio stream doesn't play an audio file: Instead, it plays a custom audio source that *you* directly provide. In other words, defining your own audio stream allows you to play from more than just a file: A sound streamed over the network, music generated by your program, an audio format that CrSFML doesn't support, etc.
 
-In fact, the [Music]({{book.api}}/Music.html) class is just a specialized audio stream that gets its audio samples from a file. 
+In fact, the [Music]({{book.api}}/Music.html) class is just a specialized audio stream that gets its audio samples from a file.
 
-Since we're talking about *streaming*, we'll deal with audio data that cannot be loaded entirely in memory, and will instead be loaded in small chunks while it is being played. If your sound can be loaded completely and can fit in memory, then audio streams won't help you: Just load the audio data into a [SoundBuffer]({{book.api}}/SoundBuffer.html) and use a regular [Sound]({{book.api}}/Sound.html) to play it. 
+Since we're talking about *streaming*, we'll deal with audio data that cannot be loaded entirely in memory, and will instead be loaded in small chunks while it is being played. If your sound can be loaded completely and can fit in memory, then audio streams won't help you: Just load the audio data into a [SoundBuffer]({{book.api}}/SoundBuffer.html) and use a regular [Sound]({{book.api}}/Sound.html) to play it.
 
-## sf::SoundStream
+## SF::SoundStream
 
-In order to define your own audio stream, you need to inherit from the [SoundStream]({{book.api}}/SoundStream.html) abstract base class. There are two virtual functions to override in your derived class: `onGetData` and `onSeek`. 
+In order to define your own audio stream, you need to inherit from the [SoundStream]({{book.api}}/SoundStream.html) abstract base class. There are two virtual functions to override in your derived class: `on_get_data` and `on_seek`.
 
-```
-class MyAudioStream : public sf::SoundStream
-{
-    virtual bool onGetData(Chunk& data);
+```ruby
+class MyAudioStream < SF::SoundStream
+    def on_get_data(data)
+        true
+    end
 
-    virtual void onSeek(sf::Time timeOffset);
-};
+    def on_seek(time_offset)
+    end
```

Comment:
Whenever there is subclassing involved, be wary that there may be no nice wrapper implemented for it, and instead just CSFML showing its ugly face. Skip this.